core/src/test/scala/org/apache/spark/scheduler/DAGSchedulerSuite.scala

```diff
@@ -800,6 +800,50 @@ class DAGSchedulerSuite
     assertDataStructuresEmpty()
   }
 
+  test("shuffle with reducer locality") {
```

Comment:
ok last supernit: can you make this name a description of what should happen? "Reduce tasks should be placed locally with map output" and then for the below test "Reduce task locality preferences should only include machine with largest map output" or something?